internal/service/kms/key_policy_attachment.go

```diff
@@ -0,0 +1,144 @@
+package kms
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/structure"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+// @SDKResource("aws_kms_key_policy_attachment")
+func ResourceKeyPolicyAttachment() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceKeyPolicyAttachmentCreate,
+		ReadWithoutTimeout:   resourceKeyPolicyAttachmentRead,
+		UpdateWithoutTimeout: resourceKeyPolicyAttachmentUpdate,
+		DeleteWithoutTimeout: resourceKeyPolicyAttachmentDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"bypass_policy_lockout_safety_check": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"key_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 2048),
+			},
+			"policy": {
+				Type:                  schema.TypeString,
+				Required:              true,
+				DiffSuppressFunc:      verify.SuppressEquivalentPolicyDiffs,
+				DiffSuppressOnRefresh: true,
+				ValidateFunc:          validation.StringIsJSON,
+				StateFunc: func(v interface{}) string {
+					json, _ := structure.NormalizeJsonString(v)
+					return json
+				},
+			},
+		},
+	}
+}
+
+func resourceKeyPolicyAttachmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	conn := meta.(*conns.AWSClient).KMSConn()
+
+	if err := updateKeyPolicy(ctx, conn, d.Get("key_id").(string), d.Get("policy").(string), d.Get("bypass_policy_lockout_safety_check").(bool)); err != nil {
```

Comment:
We should ask Hashicorp if it's acceptable to call this method on the key resource, or if we should extract it to a utility file. @ewbankkit

Reply:
Fine to keep as-is.